SwarmCallback and additional doc updates

RadhakrishnaJ opened the pull request:
1. SwarmCallback api updated to include additional parameters.
2. Added FAQ on how to know SN is successfully started.
3. Added Architecture in prerequisite.

suresh-ls commented on docs/ml_algorithm.md:
```diff
@@ -42,35 +42,42 @@ To convert a ML program into a Swarm Learning ML program:
-    ON.
+    Allowed values: ['inactive', 'snapshot', 'active'].
+    - inactive – Node does not contribute its weights in the merge process but participates as non-contributing peer in the merge process.
+    - snapshot – Node always checks in the weights that it had when it reached the stopping criterion, it will not accept merged weights. 
```
For uniformity with previous point:
change "checks in" -> "contributes"

RadhakrishnaJ replied:
updated wording.

sathya-SL commented on docs/ml_algorithm.md:
```diff
@@ -42,35 +42,42 @@ To convert a ML program into a Swarm Learning ML program:
+-   'min_peers' specifies the minimum number of network peers required to
+    synchronize the insights.
+-   'use_adaptive_sync' specifies whether the *adaptive sync interval*
     feature should be used for tuning the sync interval. This feature is
```
should be used for tuning the sync interval. -> should be used for automatically generating the sync interval.

RadhakrishnaJ replied:
Done

sathya-SL commented on docs/ml_algorithm.md:
```diff
@@ -42,35 +42,42 @@ To convert a ML program into a Swarm Learning ML program:
+-   'val_data' specifies the dataset for generating metrics for adaptive sync logic. It can be either a (x_val, y_val) tuple or a generator. This is used
+    when use_adaptive_sync is turned ON.
+-   'val_batch_size' specifies the batch size for 'val_data' dataset. This is used when use_adaptive_sync is turned ON.
+-   'checkin_model_on_train_end' specifies the merge behaviour of a SL node *after* it has achieved stopping criterion and it is waiting for all other peers to complete their training. During this period this SL node will not train model with local data. This parameter decides the nature of the weights that this SL node will contribute to the merge process. 
```
highlight this part "SL node will not train model with local data"

RadhakrishnaJ replied:
Highlighted.

sathya-SL commented on docs/ml_algorithm.md:
```diff
@@ -42,35 +42,42 @@ To convert a ML program into a Swarm Learning ML program:
+    Allowed values: ['inactive', 'snapshot', 'active'].
+    - inactive – Node does not contribute its weights in the merge process but participates as non-contributing peer in the merge process.
+    - snapshot – Node always contributes the weights that it had when it reached the stopping criterion, it will not accept merged weights. 
+    - active – Node behaves as if it is in active training. 
```
Node behaves as if it is in active training.  "But will not train merged model with local data as mentioned above."
- make the point that even though it behaves like active training it will not actually train using local data ...

RadhakrishnaJ replied:
Done.

RadhakrishnaJ commented:
Adding following content to Master. 
SwarmCallback api updated to include additional parameters.
Added FAQ on how to know SN is successfully started.
Added Architecture in prerequisite.
